### PR TITLE
Fix Actions timing cache, fall back to cron timing if needed

### PR DESCRIPTION
**Pull request by frenchy64**
Also Close https://github.com/threatgrid/iroh/issues/4950

The original goal here was to ensure pull requests don't use cron timing information, because https://github.com/threatgrid/ctia/pull/1095 made the cron build different from normal builds.

Unfortunately, the approach in https://github.com/threatgrid/ctia/pull/1095 was flawed by using `${{ github.event_name }}` in the cache key because that makes caches for `push` and `pr` distinct, which means pull requests will never see caches generated by master (and those are the only ones reachable for the first build of a PR).

Instead, this groups pr/push timing information in the same cache via `${{ env.CTIA_TEST_SUITE }}` (set to `cron` for cron, and `ci` for pr/push builds) so pr's can see push caches again, and allows pr/pushes to use cron timing as a last resort.

<a name="qa">[§](#qa)</a> QA
============================

No QA is needed.

<a name="release-notes">[§](#release-notes)</a> Release Notes
=============================================================

```
intern: Fix Actions timing cache, fall back to cron timing if needed
```

<a name="squashed-commits">[§](#squashed-commits)</a> Squashed Commits
======================================================================

### CODE REVIEW

File: scripts/actions/setup-env.sh [hunk 1]
```diff
@@ -0,0 +1,24 @@
+#!/bin/bash
+
+set -e
```
Comment (vemv) [inline scripts/actions/setup-env.sh:3]: more flags?

**Review by vemv — APPROVED** [review]
LGTM, thanks for the explanation!

File: scripts/actions/setup-env.sh (hunk 1 above)
Comment (frenchy64) [reply scripts/actions/setup-env.sh:3]: Thanks, yes. First I'll see if I can rewrite it in bb :)

**Comment by frenchy64** [comment]
I think I found the cause of some of the symptoms of https://github.com/threatgrid/iroh/issues/4950.

It turns out that `-dev` doesn't work yet to run the tests. We were warming the Maven cache with `+test`, but some of the ci commands use `-dev,+test`. This resulted in different versions of libraries being used.

I'm surprised `:pedantic? :abort` didn't pick up the differences in versions, but I guess that would only happen if we use `:managed-dependencies` to fix our transitive dependencies.

File: scripts/actions/setup-env.sh (hunk 1 above)
Comment (frenchy64) [reply scripts/actions/setup-env.sh]: Well, there was a some bootstrapping issues between this script setting up the local binary path that `bb` is installed to, but I managed to rewrite it in `bb`.

**Pull request merged**